go/cmd/dolt/commands/dump.go

```diff
@@ -0,0 +1,291 @@
+	return "Export all tables in the working set into a file."
+}
+
+// CreateMarkdown creates a markdown file containing the help text for the command at the given path
+func (cmd DumpCmd) CreateMarkdown(wr io.Writer, commandStr string) error {
+	ap := cmd.createArgParser()
+	return CreateMarkdown(wr, cli.GetCommandDocumentation(commandStr, lsDocs, ap))
+}
+
+func (cmd DumpCmd) createArgParser() *argparser.ArgParser {
+	ap := argparser.NewArgParser()
+	//ap.ArgListHelp = append(ap.ArgListHelp, [2]string{"file", "The file being output to."})
+	ap.SupportsFlag(forceParam, "f", "If data already exists in the destination, the force flag will allow the target to be overwritten.")
+	ap.SupportsString(formatParam, "r", "result_file_type", "Define the type of the output file. Valid values are sql and csv. Defaults to sql.")
+
+	return ap
+}
+
+func importStatsCB(stats types.AppliedEditStats) {
+	noEffect := stats.NonExistentDeletes + stats.SameVal
+	total := noEffect + stats.Modifications + stats.Additions
+	displayStr := fmt.Sprintf("Rows Processed: %d, Additions: %d, Modifications: %d, Had No Effect: %d", total, stats.Additions, stats.Modifications, noEffect)
+	displayStrLen = cli.DeleteAndPrint(displayStrLen, displayStr)
+}
+
+// EventType returns the type of the event to log
+func (cmd DumpCmd) EventType() eventsapi.ClientEventType {
+	return eventsapi.ClientEventType_DUMP
+}
+
+// Exec executes the command
+func (cmd DumpCmd) Exec(ctx context.Context, commandStr string, args []string, dEnv *env.DoltEnv) int {
+	ap := cmd.createArgParser()
+	_, usage := cli.HelpAndUsagePrinters(cli.GetCommandDocumentation(commandStr, dumpDocs, ap))
+
+	root, verr := GetWorkingWithVErr(dEnv)
+	if verr != nil {
+		return HandleVErrAndExitCode(verr, usage)
+	}
+
+	tblNames, err := doltdb.GetNonSystemTableNames(ctx, root)
+	if err != nil {
+		errhand.BuildDError("error: failed to get tables").AddCause(err).Build()
+	}
+
+	if len(tblNames) == 0 {
+		cli.Println("No tables to export.")
+		return 0
+	}
+
+	exOpts, verr := parseExportArgs(ap, commandStr, args)
+	if verr != nil {
+		return HandleVErrAndExitCode(verr, usage)
+	}
+
+	ow, err := exOpts.checkOverwrite(ctx, root, dEnv.FS)
+	if err != nil {
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	}
+
+	if ow {
+		return HandleVErrAndExitCode(errhand.BuildDError("%s already exists. Use -f to overwrite.", exOpts.DestName()).Build(), usage)
+	}
+
+	// create new file
+	err = dEnv.FS.MkDirs(filepath.Dir(exOpts.DestName()))
+	if err != nil {
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	}
+
+	filePath, _ := dEnv.FS.Abs(exOpts.DestName())
+	if err != nil {
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	}
+
+	os.OpenFile(filePath, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, os.ModePerm)
+
+	//cli.Printf("Tables exporting:\n")
+	for _, tbl := range tblNames {
+		//cli.Println("\t", tbl)
+
+		exOpts.tableName = tbl
```

Comment:
Rather than setting this and src on each pass through the loop, you should have two structs:

1) One dumpOpts, which has the fields you have now
2) One tableOpts, create a new one in each pass through the loop